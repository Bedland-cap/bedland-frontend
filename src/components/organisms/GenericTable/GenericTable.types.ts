export const ColumnVariants = [
  'flatsNumber',
  'floor',
  'residents',
  'monthlyPayments',
  'lastMaintenance',
  'building',
  'flat',
  'status',
  'payment',
  'deadline',
  'payments',
  'contactDetails',
] as const;

export type ColumnVariant = (typeof ColumnVariants)[number];

export type ColumnVariantsType = {
  [key in ColumnVariant]: string;
};

export const COLUMN_VARIANTS: ColumnVariantsType = {
  flatsNumber: 'Flats Number',
  floor: 'Floor',
  residents: 'Residents',
  monthlyPayments: 'Monthly Payments',
  lastMaintenance: 'Last Maintenance',
  building: 'Building',
  flat: 'Flat',
  status: 'Status',
  payment: 'Payment',
  deadline: 'Deadline',
  payments: 'Payments',
  contactDetails: 'Contact Details',
} as const;

export type nodesTypes = {
  building?: string;
  flat?: string;
  status?: Element;
  payment?: string;
  deadline?: Date;
  flatsNumber?: number;
  floor?: number;
  residents?: string[];
  monthlyPayments?: Element;
  lastMaintenance?: Date;
  payments?: Element;
  contactDetails?: string;
  id: number;
};

export type GenericTableProps = { nodes: nodesTypes[]; isSelect: boolean };
