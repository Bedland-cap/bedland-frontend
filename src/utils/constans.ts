export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
