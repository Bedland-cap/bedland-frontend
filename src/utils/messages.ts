const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
};

export default ERROR_MESSAGES;
