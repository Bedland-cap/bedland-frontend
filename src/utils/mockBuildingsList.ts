import { Building } from 'store/reducers/building/building_slice';

const mockBuildingsList: Building[] = [
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21',
    managerId: '1',
    floors: '5',
  },
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3',
    managerId: '1',
    floors: '7',
  },
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51',
    managerId: '2',
    floors: '5',
  },
  {
    id: '4',
    name: 'Desert',
    address: 'Desert Street 2X',
    managerId: '1',
    floors: '6',
  },
  {
    id: '5',
    name: 'Lakewood',
    address: 'Lakewood Avenue 57',
    managerId: '3',
    floors: '10',
  },
];

export default mockBuildingsList;
