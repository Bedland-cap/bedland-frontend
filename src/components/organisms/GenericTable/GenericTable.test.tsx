import { renderWithProviders } from 'utils/test.utils';
import nodes from 'utils/mock/mockGenericTable';
import { mapColumnHeaders, tableTheme, columnsRenderers } from './GenericTable.utils';
import GenericTable from './GenericTable';

describe('Generic Table', () => {
  it('renders correctly with selects', () => {
    const snapshot = renderWithProviders(<GenericTable nodes={nodes} isSelect />, {});
    expect(snapshot).toMatchSnapshot();
  });
  it('renders correctly without selects', () => {
    const snapshot = renderWithProviders(<GenericTable nodes={nodes} isSelect={false} />, {});
    expect(snapshot).toMatchSnapshot();
  });

  it('mapColumnHeaders correctly creates mapped object if isSelect is true', () => {
    const columns = mapColumnHeaders(nodes[0], true);

    expect(columns[0]?.select).toBeTruthy();
    expect(columns.length).toEqual(4);
    expect(Object.keys(columns[0]).length).toEqual(3);
    expect(Object.keys(columns[1]).length).toEqual(2);
  });
  it('mapColumnHeaders correctly creates mapped object if isSelect is false', () => {
    const columns = mapColumnHeaders(nodes[0], false);

    expect(columns[0]?.select).toEqual(undefined);
    expect(columns.length).toEqual(4);
    expect(Object.keys(columns[0]).length).toEqual(2);
    expect(Object.keys(columns[1]).length).toEqual(2);
  });

  it('tableTheme correctly resolves', () => {
    expect(tableTheme(4, true).Table).toEqual(`
      --data-table-library_grid-template-columns: 2.875rem repeat(4, minmax(0,3fr));
      border-radius: 8px;
      min-width: 64.0625rem;
    `);
    expect(tableTheme(3, false).Table).toEqual(`
      --data-table-library_grid-template-columns: repeat(3,3fr);
      border-radius: 8px;
      min-width: 64.0625rem;
    `);
  });
  it('columnsRenderers resolves correctly', () => {
    expect(columnsRenderers.deadline(nodes[0])).toEqual('11/17/1995');
    expect(
      columnsRenderers.lastMaintenance({ lastMaintenance: new Date('11/17/1995'), id: 1 }),
    ).toEqual('11/17/1995');
  });
});
