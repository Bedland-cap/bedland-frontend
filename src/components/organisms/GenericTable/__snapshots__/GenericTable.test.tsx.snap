// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Generic Table > renders correctly with selects 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="table css-13g0o1m-Table"
        data-table-library_table=""
        role="grid"
      >
        <thead
          class="thead css-1dpsum1-Header"
          role="rowgroup"
        >
          <tr
            class="tr tr-header css-o9t41z-getHeaderRowContainerStyle-HeaderRow"
            data-table-library_tr-header=""
            role="rowheader"
          >
            <th
              class="th stiff css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
              data-hide="false"
              data-resize-min-width="75"
              data-table-library_th=""
              role="columnheader"
            >
              <div>
                <input
                  class="css-e0dnmk"
                  type="checkbox"
                />
              </div>
            </th>
            <th
              class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
              data-hide="false"
              data-resize-min-width="75"
              data-table-library_th=""
              role="columnheader"
            >
              <div>
                Flats Number
              </div>
            </th>
            <th
              class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
              data-hide="false"
              data-resize-min-width="75"
              data-table-library_th=""
              role="columnheader"
            >
              <div>
                Floor
              </div>
            </th>
            <th
              class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
              data-hide="false"
              data-resize-min-width="75"
              data-table-library_th=""
              role="columnheader"
            >
              <div>
                Residents
              </div>
            </th>
            <th
              class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
              data-hide="false"
              data-resize-min-width="75"
              data-table-library_th=""
              role="columnheader"
            >
              <div>
                Deadline
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="tbody css-1cjq6h-Body"
          data-table-library_body=""
        >
          <tr
            class="tr tr-body row-select row-select-clickable clickable css-182qcne-getRowContainerStyle-Row"
            data-table-library_tr-body=""
            role="row"
          >
            <td
              class="td stiff css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                <input
                  class="css-e0dnmk"
                  type="checkbox"
                />
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                #1
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                1
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                <div
                  style="font-weight: bold;"
                >
                  John Paul II
                </div>
                <div
                  style="font-weight: normal;"
                >
                  Keanu Reeves
                </div>
                <div
                  style="font-weight: normal;"
                >
                  Arnold Schwarzeneger
                </div>
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                11/17/1995
              </div>
            </td>
          </tr>
          <tr
            class="tr tr-body row-select row-select-clickable clickable css-182qcne-getRowContainerStyle-Row"
            data-table-library_tr-body=""
            role="row"
          >
            <td
              class="td stiff css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                <input
                  class="css-e0dnmk"
                  type="checkbox"
                />
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                #2
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                2
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                <div
                  style="font-weight: bold;"
                >
                  John Paul II
                </div>
                <div
                  style="font-weight: normal;"
                >
                  Keanu Reeves
                </div>
                <div
                  style="font-weight: normal;"
                >
                  Arnold Schwarzeneger
                </div>
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                11/17/1995
              </div>
            </td>
          </tr>
          <tr
            class="tr tr-body row-select row-select-clickable clickable css-182qcne-getRowContainerStyle-Row"
            data-table-library_tr-body=""
            role="row"
          >
            <td
              class="td stiff css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                <input
                  class="css-e0dnmk"
                  type="checkbox"
                />
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                #3
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                3
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                <div
                  style="font-weight: bold;"
                >
                  John Paul II
                </div>
                <div
                  style="font-weight: normal;"
                >
                  Keanu Reeves
                </div>
                <div
                  style="font-weight: normal;"
                >
                  Arnold Schwarzeneger
                </div>
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                11/17/1995
              </div>
            </td>
          </tr>
        </tbody>
      </table>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": <div>
    <table
      class="table css-13g0o1m-Table"
      data-table-library_table=""
      role="grid"
    >
      <thead
        class="thead css-1dpsum1-Header"
        role="rowgroup"
      >
        <tr
          class="tr tr-header css-o9t41z-getHeaderRowContainerStyle-HeaderRow"
          data-table-library_tr-header=""
          role="rowheader"
        >
          <th
            class="th stiff css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
            data-hide="false"
            data-resize-min-width="75"
            data-table-library_th=""
            role="columnheader"
          >
            <div>
              <input
                class="css-e0dnmk"
                type="checkbox"
              />
            </div>
          </th>
          <th
            class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
            data-hide="false"
            data-resize-min-width="75"
            data-table-library_th=""
            role="columnheader"
          >
            <div>
              Flats Number
            </div>
          </th>
          <th
            class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
            data-hide="false"
            data-resize-min-width="75"
            data-table-library_th=""
            role="columnheader"
          >
            <div>
              Floor
            </div>
          </th>
          <th
            class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
            data-hide="false"
            data-resize-min-width="75"
            data-table-library_th=""
            role="columnheader"
          >
            <div>
              Residents
            </div>
          </th>
          <th
            class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
            data-hide="false"
            data-resize-min-width="75"
            data-table-library_th=""
            role="columnheader"
          >
            <div>
              Deadline
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="tbody css-1cjq6h-Body"
        data-table-library_body=""
      >
        <tr
          class="tr tr-body row-select row-select-clickable clickable css-182qcne-getRowContainerStyle-Row"
          data-table-library_tr-body=""
          role="row"
        >
          <td
            class="td stiff css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              <input
                class="css-e0dnmk"
                type="checkbox"
              />
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              #1
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              1
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              <div
                style="font-weight: bold;"
              >
                John Paul II
              </div>
              <div
                style="font-weight: normal;"
              >
                Keanu Reeves
              </div>
              <div
                style="font-weight: normal;"
              >
                Arnold Schwarzeneger
              </div>
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              11/17/1995
            </div>
          </td>
        </tr>
        <tr
          class="tr tr-body row-select row-select-clickable clickable css-182qcne-getRowContainerStyle-Row"
          data-table-library_tr-body=""
          role="row"
        >
          <td
            class="td stiff css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              <input
                class="css-e0dnmk"
                type="checkbox"
              />
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              #2
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              2
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              <div
                style="font-weight: bold;"
              >
                John Paul II
              </div>
              <div
                style="font-weight: normal;"
              >
                Keanu Reeves
              </div>
              <div
                style="font-weight: normal;"
              >
                Arnold Schwarzeneger
              </div>
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              11/17/1995
            </div>
          </td>
        </tr>
        <tr
          class="tr tr-body row-select row-select-clickable clickable css-182qcne-getRowContainerStyle-Row"
          data-table-library_tr-body=""
          role="row"
        >
          <td
            class="td stiff css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              <input
                class="css-e0dnmk"
                type="checkbox"
              />
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              #3
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              3
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              <div
                style="font-weight: bold;"
              >
                John Paul II
              </div>
              <div
                style="font-weight: normal;"
              >
                Keanu Reeves
              </div>
              <div
                style="font-weight: normal;"
              >
                Arnold Schwarzeneger
              </div>
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              11/17/1995
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Generic Table > renders correctly without selects 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="table css-omp3au-Table"
        data-table-library_table=""
        role="grid"
      >
        <thead
          class="thead css-1dpsum1-Header"
          role="rowgroup"
        >
          <tr
            class="tr tr-header css-pdmx8e-getHeaderRowContainerStyle-HeaderRow"
            data-table-library_tr-header=""
            role="rowheader"
          >
            <th
              class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
              data-hide="false"
              data-resize-min-width="75"
              data-table-library_th=""
              role="columnheader"
            >
              <div>
                Flats Number
              </div>
            </th>
            <th
              class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
              data-hide="false"
              data-resize-min-width="75"
              data-table-library_th=""
              role="columnheader"
            >
              <div>
                Floor
              </div>
            </th>
            <th
              class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
              data-hide="false"
              data-resize-min-width="75"
              data-table-library_th=""
              role="columnheader"
            >
              <div>
                Residents
              </div>
            </th>
            <th
              class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
              data-hide="false"
              data-resize-min-width="75"
              data-table-library_th=""
              role="columnheader"
            >
              <div>
                Deadline
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="tbody css-1cjq6h-Body"
          data-table-library_body=""
        >
          <tr
            class="tr tr-body css-1u6rcw7-getRowContainerStyle-Row"
            data-table-library_tr-body=""
            role="row"
          >
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                #1
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                1
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                <div
                  style="font-weight: bold;"
                >
                  John Paul II
                </div>
                <div
                  style="font-weight: normal;"
                >
                  Keanu Reeves
                </div>
                <div
                  style="font-weight: normal;"
                >
                  Arnold Schwarzeneger
                </div>
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                11/17/1995
              </div>
            </td>
          </tr>
          <tr
            class="tr tr-body css-1u6rcw7-getRowContainerStyle-Row"
            data-table-library_tr-body=""
            role="row"
          >
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                #2
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                2
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                <div
                  style="font-weight: bold;"
                >
                  John Paul II
                </div>
                <div
                  style="font-weight: normal;"
                >
                  Keanu Reeves
                </div>
                <div
                  style="font-weight: normal;"
                >
                  Arnold Schwarzeneger
                </div>
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                11/17/1995
              </div>
            </td>
          </tr>
          <tr
            class="tr tr-body css-1u6rcw7-getRowContainerStyle-Row"
            data-table-library_tr-body=""
            role="row"
          >
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                #3
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                3
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                <div
                  style="font-weight: bold;"
                >
                  John Paul II
                </div>
                <div
                  style="font-weight: normal;"
                >
                  Keanu Reeves
                </div>
                <div
                  style="font-weight: normal;"
                >
                  Arnold Schwarzeneger
                </div>
              </div>
            </td>
            <td
              class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
              data-table-library_td=""
              role="gridcell"
            >
              <div>
                11/17/1995
              </div>
            </td>
          </tr>
        </tbody>
      </table>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": <div>
    <table
      class="table css-omp3au-Table"
      data-table-library_table=""
      role="grid"
    >
      <thead
        class="thead css-1dpsum1-Header"
        role="rowgroup"
      >
        <tr
          class="tr tr-header css-pdmx8e-getHeaderRowContainerStyle-HeaderRow"
          data-table-library_tr-header=""
          role="rowheader"
        >
          <th
            class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
            data-hide="false"
            data-resize-min-width="75"
            data-table-library_th=""
            role="columnheader"
          >
            <div>
              Flats Number
            </div>
          </th>
          <th
            class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
            data-hide="false"
            data-resize-min-width="75"
            data-table-library_th=""
            role="columnheader"
          >
            <div>
              Floor
            </div>
          </th>
          <th
            class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
            data-hide="false"
            data-resize-min-width="75"
            data-table-library_th=""
            role="columnheader"
          >
            <div>
              Residents
            </div>
          </th>
          <th
            class="th css-12ezvgm-HEADER_CELL_CONTAINER_STYLE-HeaderCell"
            data-hide="false"
            data-resize-min-width="75"
            data-table-library_th=""
            role="columnheader"
          >
            <div>
              Deadline
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="tbody css-1cjq6h-Body"
        data-table-library_body=""
      >
        <tr
          class="tr tr-body css-1u6rcw7-getRowContainerStyle-Row"
          data-table-library_tr-body=""
          role="row"
        >
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              #1
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              1
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              <div
                style="font-weight: bold;"
              >
                John Paul II
              </div>
              <div
                style="font-weight: normal;"
              >
                Keanu Reeves
              </div>
              <div
                style="font-weight: normal;"
              >
                Arnold Schwarzeneger
              </div>
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              11/17/1995
            </div>
          </td>
        </tr>
        <tr
          class="tr tr-body css-1u6rcw7-getRowContainerStyle-Row"
          data-table-library_tr-body=""
          role="row"
        >
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              #2
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              2
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              <div
                style="font-weight: bold;"
              >
                John Paul II
              </div>
              <div
                style="font-weight: normal;"
              >
                Keanu Reeves
              </div>
              <div
                style="font-weight: normal;"
              >
                Arnold Schwarzeneger
              </div>
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              11/17/1995
            </div>
          </td>
        </tr>
        <tr
          class="tr tr-body css-1u6rcw7-getRowContainerStyle-Row"
          data-table-library_tr-body=""
          role="row"
        >
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              #3
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              3
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              <div
                style="font-weight: bold;"
              >
                John Paul II
              </div>
              <div
                style="font-weight: normal;"
              >
                Keanu Reeves
              </div>
              <div
                style="font-weight: normal;"
              >
                Arnold Schwarzeneger
              </div>
            </div>
          </td>
          <td
            class="td css-1jvtg6b-CELL_CONTAINER_STYLE-Cell"
            data-table-library_td=""
            role="gridcell"
          >
            <div>
              11/17/1995
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
